src/lcm/create_params.py

```diff
@@ -1,138 +1,176 @@
 """Create a parameters for a model specification."""
 
 import inspect
+from typing import Any
 
-import numpy as np
+import jax.numpy as jnp
+import pandas as pd
+from jax import Array
 
 
-def create_params(user_model, variable_info, grids):
-    """Get parameters from a model specification.
+def create_params_template(
+    model_spec: dict[str, Any],
+    variable_info: pd.DataFrame,
+    grids: dict[str, Array],
+    default_params: dict[str, float] | None = None,
+) -> dict[str, dict[str, Any] | float]:
+    """Create parameter template from a model specification.
 
     Args:
-        user_model (dict): A model specification. Has keys
+        model_spec (dict): A model specification provided by the user. Has keys
             - "functions": A dictionary of functions used in the model.
             - "choices": A dictionary of choice variables.
             - "states": A dictionary of state variables.
             - "n_periods": Number of periods in the model (int).
             - "shocks": A dictionary of shock variables (optional).
         variable_info (pd.DataFrame): A dataframe with information about the variables.
-        grids (dict): A dictionary of grids.
+        grids (dict): A dictionary of grids retrieved from model_spec.
+        default_params (dict): A dictionary of default parameters. Default is None. If
+            None, the default {"beta": np.nan} is used. For other lifetime reward
+            objectives, additional parameters may be required, for example {"beta":
+            np.nan, "delta": np.nan} for beta-delta discounting.
 
     Returns:
-        dict: A dictionary of model parameters.
+        dict: A (nested) dictionary of model parameters.
 
     """
-    params = _create_standard_params() | _create_function_params(user_model)
+    if default_params is None:
+        # The default lifetime reward objective in LCM is expected discounted utility.
+        # For this objective the only additional parameter is the discounting rate beta.
+        default_params = {"beta": jnp.nan}
+
+    params_template = default_params | _create_function_params(model_spec)
 
     if variable_info["is_stochastic"].any():
-        params["shocks"] = _create_shock_params(
-            user_model,
+        params_template["shocks"] = _create_stochastic_transition_params(
+            model_spec=model_spec,
             variable_info=variable_info,
             grids=grids,
         )
 
-    return params
+    return params_template
 
 
-def _create_function_params(model):
+def _create_function_params(model_spec: dict[str, Any]) -> dict[str, dict[str, Any]]:
     """Get function parameters from a model specification.
```

Comment:
Would be good to explain a little bit what exactly is being done. IIUC:
- Look at grand DAG
- Remove all arguments that are states, choices, user-provided functions, etc (i.e., things LCM knows from the model spec)
- Assume everything else is a parameter that will need to be provided by the user.